tests/jax/components/executing/observing_test.py

```diff
@@ -0,0 +1,272 @@
+        mock_executor_without_adder: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe_first(
+        executor=mock_executor_without_adder
+    )
+
+    assert not mock_executor_without_adder.store.adder
+
+
+def test_on_execution_observe_first(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor: MockExecutor,
+) -> None:
+    """Test on_execution_observe_first method from FeedForwardExecutorObserve
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe_first(executor=mock_executor)
+
+    assert not mock_executor.store.network_int_keys_extras == None
+    assert (
+        mock_executor.store.extras["network_int_keys"]
+        == mock_executor.store.network_int_keys_extras
+    )
+    assert mock_executor.store.adder.parm == "after_add_first"
+
+
+# test on_execution_observe
+def test_on_execution_observe_without_adder(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor_without_adder: Executor,
+) -> None:
+    """Test entering executor without store.adder
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor_without_adder: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe(
+        executor=mock_executor_without_adder
+    )
+
+    assert not mock_executor_without_adder.store.adder
+
+
+def test_on_execution_observe(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor: MockExecutor,
+) -> None:
+    """Test on_execution_observe method from FeedForwardExecutorObserve
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe(executor=mock_executor)
+
+    for agent in mock_executor.store.policies_info.keys():
+        assert mock_executor.store.next_extras["policy_info"][
+            agent
+        ] == "policy_info_" + str(agent)
+    assert (
+        mock_executor.store.next_extras["network_int_keys"]
+        == mock_executor.store.network_int_keys_extras
+    )
+    assert mock_executor.store.adder.parm == "after_add"
+
+
+# test on_execution_update
+def test_on_execution_update_without_parameter_client(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor_without_parameter_client: Executor,
+) -> None:
+    """Test entering executor without store.executor_parameter_client
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor_without_parameter_client: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_update(
+        executor=mock_executor_without_parameter_client
+    )
+
+    assert not mock_executor_without_parameter_client.store.executor_parameter_client
```

Comment:
Again, not the correct assertion. You want to assert that `get_async()` was not called.

Reply:
I removed the test of get_async since we agree on that in one of the previous standups